src/app/client/src/app/modules/dashboard/components/filter/filter.component.ts

```diff
@@ -0,0 +1,237 @@
+import { Component, OnInit, EventEmitter, Input, Output, OnDestroy, ViewChild, ElementRef, ChangeDetectorRef } from '@angular/core';
+import { IInteractEventObject } from '@sunbird/telemetry';
+import { ResourceService } from '@sunbird/shared';
+import { FormGroup, FormBuilder } from '@angular/forms';
+import * as _ from 'lodash-es';
+import * as moment from 'moment';
+import { ActivatedRoute } from '@angular/router';
+import { Subscription, Subject } from 'rxjs';
+import { distinctUntilChanged, map, debounceTime, takeUntil } from 'rxjs/operators';
+
+
+@Component({
+  selector: 'app-filter',
+  templateUrl: './filter.component.html',
+  styleUrls: ['./filter.component.scss']
+})
+export class FilterComponent implements OnInit, OnDestroy {
+
+  @Input() hideElements: boolean = false;
+  @Input() chartData: any;
+  @Input() filters: any;
+  @Input() telemetryInteractObject: IInteractEventObject;
+  @Input() chartType: any;
```

Comment:
This component's job is only meant to filter out the data. Why do you need to pass the chartType and graph stats input in this component. Please keep these logics inside the chart component only...

Reply:
resolved